### PR TITLE
refactor login.js and login.liquid to remove jquery dependency

### LINKED ISSUE #138: drop jquery dependency in login.js
#122 Makes some great changes to the theme addresses, and also removes jQuery. I'm not sure if dropping jQuery support is a Long Term Goal of this repo?

The login and addresses functionality are so tied together so if given the go ahead I'd like to remove the jquery dependency that relates to the `login.js` and subsequently `login.liquid`

From my understanding once this PR request is merged there is just `jquery` left in `product.js` an `gift_card.js` which I am also happy to tackle down the line.

I know this also makes no difference if someone installs an App that has jquery as a dependency, but would maybe be nice to give the shopify/slate community the ability to not have to use it if they don't want to.

### PR DESCRIPTION
**Pull request by isaac-martin**
This PR closes #138 
Inspired by the excellent work done on #122 

### Changes
- [x] update selectors in `login.liquid` to use data-attributes
- [x]  update `login.js` to reference new selectors and replace jquery with vanilla js
  
### How to test

- Preview with https://smkktest123.myshopify.com
- Make a user account
- Check you can login as intended
- Check you can trigger password reset and a confirmation message is shown

### Expectations:

- [x] It functions as before
- [x] It follows the same code style that we are using on `addresses.js`

### CODE REVIEW

**Comment by t-kelly** [comment]
@isaac-martin can we please wrap this PR up?

**Comment by dan-gamble** [comment]
I'll be happy to pick this one up if @isaac-martin finds himself too busy 🤙

**Comment by t-kelly** [comment]
If @isaac-martin doesn't want to take all the glory, feel free to seize it @dan-gamble!

**Comment by dan-gamble** [comment]
I'll give him a couple of days then if we don't hear anything i'll help co-author this. Anything i can do to get rid of jQuery i'm game for 😄

**Comment by isaac-martin** [comment]
Sorry - will finalize tomorrow @dan-gamble and @t-kelly Have been on vacation until today

File: src/scripts/templates/customers/login.js [hunk 1]
```diff
@@ -6,11 +6,12 @@
  * @namespace password
  */
 
-import $ from 'jquery';
-
 const selectors = {
-  recoverPasswordForm: '#RecoverPassword',
-  hideRecoverPasswordLink: '#HideRecoverPasswordLink',
+  recoverPasswordFormTriggers: document.querySelectorAll('[data-recover-toggle]'),
```
[the remark this answers is not in the document]
Comment (isaac-martin) [reply src/scripts/templates/customers/login.js]: Adjusted in latest commit

File: src/scripts/templates/customers/login.js [hunk 2]
```diff
@@ -31,29 +32,27 @@ function checkUrlHash() {
  *  Show/Hide recover password form
  */
 function toggleRecoverPasswordForm() {
-  $('#RecoverPasswordForm').toggleClass('hide');
-  $('#CustomerLoginForm').toggleClass('hide');
+  selectors.recoverPasswordForm.classList.toggle('hide');
```
[the remark this answers is not in the document]
Comment (isaac-martin) [reply src/scripts/templates/customers/login.js]: Adjusted in latest commit

File: src/scripts/templates/customers/login.js [hunk 3]
```diff
@@ -31,29 +32,27 @@ function checkUrlHash() {
...
+  selectors.loginForm.classList.toggle('hide');
```
[the remark this answers is not in the document]
Comment (isaac-martin) [reply src/scripts/templates/customers/login.js]: Adjusted in latest commit

**Comment by dan-gamble** [comment]
No worries @isaac-martin, just wanted to help out if i was needed. Hope you enjoyed your holiday 😊.

Looking forward to having this one in!

File: src/scripts/templates/customers/login.js [hunk 4]
```diff
@@ -31,29 +32,27 @@ function checkUrlHash() {
...
+  document.querySelector(selectors.loginForm).classList.toggle('hide');
 }
 
 /**
  *  Show reset password success message
  */
 function resetPasswordSuccess() {
-  const $formState = $('.reset-password-success');
+  // check if reset password form was
+  // successfully submited and show success message.
```
Comment (chrisberthe) [inline src/scripts/templates/customers/login.js]:
```suggestion
  // successfully submitted and show success message.
```

File: src/templates/customers/login.liquid [hunk 5]
```diff
@@ -71,7 +71,7 @@
 
     <input type="submit" class="button" value="{{ 'customer.recover_password.submit' | t }}">
 
-    <button type="button" id="HideRecoverPasswordLink">
+    <button type="button" data-recover-toggle >
```
Comment (chrisberthe) [inline src/templates/customers/login.liquid]:
```suggestion
    <button type="button" data-recover-toggle>
```

File: src/templates/customers/login.liquid [hunk 6]
```diff
@@ -41,21 +41,21 @@
...
     {% if form.password_needed %}
-      <a href="#recover" id="RecoverPassword">{{ 'customer.login.forgot_password' | t }}</a>
+      <a href="#recover" data-recover-toggle>{{ 'customer.login.forgot_password' | t }}</a>
     {% endif %}
   {% endform %}
 
 </div>
 
-<div id="RecoverPasswordForm" class="hide">
+<div data-recover-form class="hide">
```
Comment (chrisberthe) [inline src/templates/customers/login.liquid]:
```suggestion
<div class="hide" data-recover-form>
```

We tend to add data attributes last in attribute order.

**Comment by matjazzorko** [comment]
I had an issue with document.querySelector() on line 55, shouldn't it be document.querySelectorAll() ?

**Comment by t-kelly** [comment]
@matjazzorko open a PR! 😀

**Pull request merged**